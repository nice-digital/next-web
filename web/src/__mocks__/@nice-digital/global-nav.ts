// Mock global nav as an empty function as there's so much markup it causes loads of noise
// when tests fail and React Testing Library lists nodes for debugging purposes.
// Use jest.requireActual("@nice-digital/global-nav"); if you need the original implementation for testing purposes

import { FC } from "react";

export const Header: FC = jest.fn().mockImplementation(() => null);

export const Footer: FC = jest.fn().mockImplementation(() => null);
