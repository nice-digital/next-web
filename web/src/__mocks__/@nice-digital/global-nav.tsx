// Mock global nav as an empty function as there's so much markup it causes loads of noise
// when tests fail and React Testing Library lists nodes for debugging purposes.
// Use jest.requireActual("@nice-digital/global-nav"); if you need the original implementation for testing purposes

import { FC, PropsWithChildren } from "react";

export const Header: FC = jest.fn().mockImplementation(() => null);

export const Footer: FC = jest.fn().mockImplementation(() => null);

// Mock a simpler version of main (without back to top) to avoid noise in tests and snapshots
export const Main = jest.fn(
	({ children }: PropsWithChildren<Record<string, never>>) => (
		<main>{children}</main>
	)
);
