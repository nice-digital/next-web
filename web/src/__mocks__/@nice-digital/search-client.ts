import mockPublishedJsonResponse from "@/mockData/search/guidance-published.json";

const searchClient = jest.requireActual("@nice-digital/search-client");

module.exports = {
	...searchClient,
	search: jest.fn().mockResolvedValue(mockPublishedJsonResponse),
};
