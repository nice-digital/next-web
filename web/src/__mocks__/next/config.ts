import nextConfig from "./../../../next.config";

const getConfig = (): unknown => ({
	publicRuntimeConfig: nextConfig.publicRuntimeConfig,
	serverRuntimeConfig: nextConfig.serverRuntimeConfig,
});

export default getConfig;
