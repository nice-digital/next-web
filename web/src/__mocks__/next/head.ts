/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { FC } from "react";
import { renderToString } from "react-dom/server";

interface HeadProps {
	children: Parameters<typeof renderToString>[0];
}

/**
 * Mocking next/head allows us to use it in our tests to assert against meta tags/titles
 * See https://edibleco.de/3dyJuUl
 */
const Head: FC<HeadProps> = ({ children }: HeadProps) => {
	if (children) {
		global.document.head.insertAdjacentHTML(
			"afterbegin",
			renderToString(children) || ""
		);
	}
	return null;
};

export default Head;
