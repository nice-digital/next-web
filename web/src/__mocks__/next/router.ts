export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
