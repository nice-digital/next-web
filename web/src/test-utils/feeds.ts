import fs from "fs/promises";
import path from "path";

import MockAdapter from "axios-mock-adapter";

import { client } from "../feeds";

const mockDataBaseDirectory = path.resolve(
		__dirname,
		"../",
		"__mocks__",
		"__data__"
	),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	publicationsBaseUrl = new URL(process.env.SERVER_FEEDS_PUBLICATIONS_ORIGIN!),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	inDevBaseUrl = new URL(process.env.SERVER_FEEDS_INDEV_ORIGIN!);

export const axiosJSONMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

/**
 * Adds default JSON endpoint mocks for publications and indev feeds.
 *
 * Maps the incoming request URL to static JSON files on the file system in the mock data folder
 */
export const addDefaultJSONFeedMocks = (): MockAdapter =>
	axiosJSONMock.onGet().reply(async (config) => {
		if (!config.url) throw Error(`URL was empty in Axios mock`);

		const { host, pathname } = new URL(config.url),
			isHTML = config.headers?.Accept === "text/html";

		let filePath = mockDataBaseDirectory;

		if (host === publicationsBaseUrl.host) {
			filePath = path.join(filePath, "publications");
		} else if (host === inDevBaseUrl.host) {
			filePath = path.join(filePath, "inDev");
		} else
			throw Error(
				`Unknown URL host for mocking. Expected either ${publicationsBaseUrl.host} or ${inDevBaseUrl.host} but got ${host}`
			);

		filePath = path.join(
			filePath,
			pathname.toLowerCase() + (isHTML ? ".html" : ".json")
		);

		try {
			await fs.access(filePath);
		} catch {
			throw Error(`Could not find mock JSON file with path ${filePath}`);
		}

		const fileContents = await fs.readFile(filePath, "utf8");

		return [
			200,
			isHTML ? fileContents : !fileContents ? "" : JSON.parse(fileContents),
		];
	});
