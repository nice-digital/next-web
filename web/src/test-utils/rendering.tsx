/* eslint-disable import/export, import/no-unresolved -- We're deliberately overrind the default render from @testing-library with a custom render method */
import {
	render as testingLibrRender,
	RenderResult,
} from "@testing-library/react";
import { HeadManagerContext } from "next/dist/shared/lib/head-manager-context.shared-runtime";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { NextRouter } from "next/router";
import { ReactElement, useEffect } from "react";
import { renderToString } from "react-dom/server";

import NextWebApp from "../pages/_app";

import type { AppProps } from "next/app";

/**
 *
 */
export const mockRouter: NextRouter = {
	basePath: "",
	pathname: "/",
	route: "/",
	asPath: "/",
	query: {},
	push: jest.fn(),
	replace: jest.fn(),
	reload: jest.fn(),
	back: jest.fn(),
	forward: jest.fn(),
	prefetch: jest.fn(),
	beforePopState: jest.fn(),
	events: {
		on: jest.fn(),
		off: jest.fn(),
		emit: jest.fn(),
	},
	isFallback: false,
	isLocaleDomain: false,
	isPreview: false,
	isReady: false,
};

/**
 * Convenience method for getting a mock router with the given properties.
 *
 * @example
 * 	getMockRouter({ pathname: "/about" })
 *
 * @param routerParams Optional router properties that get merged with the default mock router
 * @returns A new mock router object
 */
export const getMockRouter = (
	routerParams?: Partial<NextRouter>
): NextRouter => ({
	...mockRouter,
	...routerParams,
});

/**
 * Collect tags from next/head: https://edibleco.de/3Anu92S
 */
const HeadProvider: React.FC = ({ children }) => {
	let head: JSX.Element[];

	useEffect(() => {
		global.document.head.insertAdjacentHTML(
			"afterbegin",
			renderToString(<>{head}</>) || ""
		);
	});

	return (
		<HeadManagerContext.Provider
			value={{
				updateHead: (state) => {
					head = state;
				},
				mountedInstances: new Set(),
			}}
		>
			{children}
		</HeadManagerContext.Provider>
	);
};

export type GetWrapperOptions = {
	router: NextRouter;
	wrapper: RenderOptions["wrapper"];
};
const getAppWrapper =
	({ router, wrapper: Wrapper }: GetWrapperOptions) =>
	({ children }: { children: ReactElement }) =>
		(
			<HeadProvider>
				<RouterContext.Provider value={router}>
					<NextWebApp
						Component={() => (
							<>{Wrapper ? <Wrapper>{children}</Wrapper> : children}</>
						)}
						pageProps={{}}
						router={router as AppProps["router"]}
					/>
				</RouterContext.Provider>
			</HeadProvider>
		);

export type RenderOptions = Parameters<typeof testingLibrRender>[1] & {
	router?: Partial<NextRouter>;
};

/**
 * Custom render method the given component, wrapped in `NextWebApp` and a `HeadProvider`.
 *
 * @see https://testing-library.com/docs/react-testing-library/setup/#custom-render
 *
 * @param ui The element to render
 * @param options Optional render options that get passed to underlying React Testing Lib render method
 * @returns The render result
 */
const customRender = (
	ui: ReactElement,
	{ wrapper, router, ...options }: RenderOptions = {}
): RenderResult =>
	testingLibrRender(ui, {
		wrapper: getAppWrapper({ router: getMockRouter(router), wrapper }),
		...options,
	});

export * from "@testing-library/react";
export { customRender as render };
