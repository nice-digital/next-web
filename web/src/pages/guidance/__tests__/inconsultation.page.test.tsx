import mockDate from "mockdate";
import { useRouter } from "next/router";

import { SearchResultsSuccess, SearchUrl } from "@nice-digital/search-client";

import { render } from "@/test-utils";

import sampleData from "../../../__mocks__/__data__/search/guidance-inconsultation.json";
import InConsultationPage from "../inconsultation.page";

(useRouter as jest.Mock).mockImplementation(() => ({
	route: "/",
	pathname: "",
	query: "",
	asPath: "",
	push: jest.fn(),
}));

describe("/guidance/inconsultation", () => {
	it("should match the snapshot", () => {
		mockDate.set("2020-11-22");

		expect(
			render(
				<InConsultationPage
					activeModifiers={[]}
					results={sampleData as unknown as SearchResultsSuccess}
					searchUrl={{ route: "/guidance/inconsultation" } as SearchUrl}
				/>
			).container
		).toMatchSnapshot();
	});
});
