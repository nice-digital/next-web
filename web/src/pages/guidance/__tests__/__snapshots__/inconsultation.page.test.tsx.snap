// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/guidance/inconsultation should match the snapshot 1`] = `
<div>
  <main
    id="content-start"
  >
    <div
      class="container"
    >
      <nav
        aria-label="Breadcrumbs"
        role="navigation"
      >
        <ol
          class="breadcrumbs"
        >
          <li
            class="breadcrumbs__crumb"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="breadcrumbs__crumb"
          >
            <a
              href="/guidance"
            >
              NICE guidance
            </a>
          </li>
          <li
            class="breadcrumbs__crumb"
          >
            <span>
              In consultation
            </span>
          </li>
        </ol>
      </nav>
      <script
        type="application/ld+json"
      >
        {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/guidance",
        "name": "NICE guidance"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "In consultation"
      }
    }
  ]
}
      </script>
      <div
        class="page-header"
      >
        <h1
          class="page-header__heading"
        >
          <span
            class="page-header__pre-heading"
          >
            Guidance and quality standards 
          </span>
          In consultation
        </h1>
        <p
          class="page-header__lead"
        >
          <a
            class="skipLink"
            href="#filters"
          >
            Skip to filters
          </a>
          <a
            class="skipLink"
            href="#results"
          >
            Skip to results
          </a>
        </p>
      </div>
      <nav
        aria-label="Stages of guidance development"
        class="horizontal-nav"
      >
        <ul
          class="horizontal-nav__list"
        >
          <li
            class="horizontal-nav__item"
          >
            <a
              aria-current="false"
              class="horizontal-nav__link"
              href="/guidance/published"
            >
              Published
            </a>
          </li>
          <li
            class="horizontal-nav__item"
          >
            <a
              aria-current="false"
              class="horizontal-nav__link"
              href="/guidance/inconsultation"
            >
              In consultation
            </a>
          </li>
          <li
            class="horizontal-nav__item"
          >
            <a
              aria-current="false"
              class="horizontal-nav__link"
              href="/guidance/indevelopment"
            >
              In development
            </a>
          </li>
          <li
            class="horizontal-nav__item"
          >
            <a
              aria-current="false"
              class="horizontal-nav__link"
              href="/guidance/proposed"
            >
              Proposed
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="grid grid--loose sectionWrapper"
      >
        <section
          aria-label="Filter results"
          class="panelWrapper"
          data-g="12 md:4 lg:3"
        >
          <form
            aria-label="Filter results"
            id="filters"
          >
            <div
              class="filter-panel"
            >
              <h2
                class="filter-panel__heading"
              >
                <button
                  aria-controls="filter-panel-body"
                  aria-expanded="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="filter-panel__heading-icon filter-panel__heading-icon--expanded"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                      fill="currentColor"
                    />
                  </svg>
                  Filter
                </button>
              </h2>
              <div
                aria-hidden="false"
                class="filter-panel__body"
                id="filter-panel-body"
              >
                <a
                  class="skipLink"
                  href="#results"
                >
                  Skip to results
                </a>
                <input
                  name="ps"
                  type="hidden"
                  value=""
                />
                <input
                  name="s"
                  type="hidden"
                  value=""
                />
                <div
                  class="container"
                >
                  <label
                    class="label"
                    for="q"
                  >
                    Filter by title or keyword
                  </label>
                  <div
                    class="input"
                  >
                    <input
                      autocomplete="off"
                      class="input__input"
                      id="q"
                      name="q"
                      placeholder="E.g. 'diabetes' or 'NG28'"
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    class="btn button"
                    type="submit"
                  >
                    Filter
                  </button>
                </div>
                <div
                  class="filter-group"
                >
                  <h3
                    class="filter-group__heading"
                  >
                    <button
                      aria-controls="group-consultation-end-date"
                      aria-expanded="true"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="filter-group__heading-icon filter-group__heading-icon--expanded"
                        fill="none"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                          fill="currentColor"
                        />
                      </svg>
                      <span
                        id="group-heading-consultation-end-date"
                      >
                        Consultation end date
                         
                      </span>
                    </button>
                  </h3>
                  <fieldset
                    aria-hidden="false"
                    class="filter-group__options"
                    id="group-consultation-end-date"
                  >
                    <legend>
                      Consultation end date
                    </legend>
                    <div
                      class="ph--c pt--c"
                    >
                      <div
                        class="input"
                      >
                        <label
                          class="input__label"
                          for="from"
                        >
                          From date
                        </label>
                        <input
                          class="input__input"
                          id="from"
                          max="2030-11-22"
                          min="2020-11-22"
                          name="from"
                          pattern="[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01])"
                          placeholder="yyyy-mm-dd"
                          type="date"
                          value=""
                        />
                      </div>
                      <div
                        class="input"
                      >
                        <label
                          class="input__label"
                          for="to"
                        >
                          To date
                        </label>
                        <input
                          class="input__input"
                          id="to"
                          max="2030-11-22"
                          min="2020-11-22"
                          name="to"
                          pattern="[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01])"
                          placeholder="yyyy-mm-dd"
                          type="date"
                          value=""
                        />
                      </div>
                      <button
                        class="btn btn--secondary ml--0"
                        type="submit"
                      >
                        Apply date filters
                      </button>
                    </div>
                  </fieldset>
                </div>
                <div
                  class="filter-group"
                >
                  <h3
                    class="filter-group__heading"
                  >
                    <button
                      aria-controls="group-ndt"
                      aria-expanded="true"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="filter-group__heading-icon filter-group__heading-icon--expanded"
                        fill="none"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                          fill="currentColor"
                        />
                      </svg>
                      <span
                        id="group-heading-ndt"
                      >
                        Type
                         
                      </span>
                    </button>
                  </h3>
                  <fieldset
                    aria-hidden="false"
                    class="filter-group__options"
                    id="group-ndt"
                  >
                    <legend>
                      Type
                    </legend>
                    <label
                      class="filter-option"
                      for="filter_ndt_guidance"
                    >
                      <input
                        id="filter_ndt_guidance"
                        name="ndt"
                        title="Type - Guidance (14)"
                        type="checkbox"
                        value="Guidance"
                      />
                      <span
                        class="filter-option__text"
                      >
                        Guidance (14)
                      </span>
                    </label>
                  </fieldset>
                </div>
                <div
                  class="filter-group"
                >
                  <h3
                    class="filter-group__heading"
                  >
                    <button
                      aria-controls="group-ngt"
                      aria-expanded="false"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="filter-group__heading-icon"
                        fill="none"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                          fill="currentColor"
                        />
                      </svg>
                      <span
                        id="group-heading-ngt"
                      >
                        Guidance programme
                         
                      </span>
                    </button>
                  </h3>
                  <fieldset
                    aria-hidden="true"
                    class="filter-group__options"
                    id="group-ngt"
                  >
                    <legend>
                      Guidance programme
                    </legend>
                    <label
                      class="filter-option"
                      for="filter_ngt_diagnostics-guidance"
                    >
                      <input
                        id="filter_ngt_diagnostics-guidance"
                        name="ngt"
                        title="Guidance programme - Diagnostics guidance (1)"
                        type="checkbox"
                        value="Diagnostics guidance"
                      />
                      <span
                        class="filter-option__text"
                      >
                        Diagnostics guidance (1)
                      </span>
                    </label>
                    <label
                      class="filter-option"
                      for="filter_ngt_interventional-procedures-guidance"
                    >
                      <input
                        id="filter_ngt_interventional-procedures-guidance"
                        name="ngt"
                        title="Guidance programme - Interventional procedures guidance (6)"
                        type="checkbox"
                        value="Interventional procedures guidance"
                      />
                      <span
                        class="filter-option__text"
                      >
                        Interventional procedures guidance (6)
                      </span>
                    </label>
                    <label
                      class="filter-option"
                      for="filter_ngt_medical-technologies-guidance"
                    >
                      <input
                        id="filter_ngt_medical-technologies-guidance"
                        name="ngt"
                        title="Guidance programme - Medical technologies guidance (1)"
                        type="checkbox"
                        value="Medical technologies guidance"
                      />
                      <span
                        class="filter-option__text"
                      >
                        Medical technologies guidance (1)
                      </span>
                    </label>
                    <label
                      class="filter-option"
                      for="filter_ngt_nice-guideline"
                    >
                      <input
                        id="filter_ngt_nice-guideline"
                        name="ngt"
                        title="Guidance programme - NICE guideline (4)"
                        type="checkbox"
                        value="NICE guideline"
                      />
                      <span
                        class="filter-option__text"
                      >
                        NICE guideline (4)
                      </span>
                    </label>
                    <label
                      class="filter-option"
                      for="filter_ngt_technology-appraisal-guidance"
                    >
                      <input
                        id="filter_ngt_technology-appraisal-guidance"
                        name="ngt"
                        title="Guidance programme - Technology appraisal guidance (2)"
                        type="checkbox"
                        value="Technology appraisal guidance"
                      />
                      <span
                        class="filter-option__text"
                      >
                        Technology appraisal guidance (2)
                      </span>
                    </label>
                  </fieldset>
                </div>
              </div>
            </div>
          </form>
        </section>
        <section
          aria-labelledby="filter-summary"
          data-g="12 md:8 lg:9"
        >
          <div
            class="filter-summary"
            id="filter-summary"
          >
            <div
              class="filter-summary__count"
            >
              <h2
                class="h5 mv--0"
              >
                Showing 
                1
                 to 
                10
                 of 
                14
              </h2>
            </div>
            <p
              class="filter-summary__sort hide-print"
            >
              <span>
                <span
                  class="visually-hidden"
                >
                  Sorted by
                </span>
                 
                Date
                <span
                  class="visually-hidden"
                >
                  .
                </span>
              </span>
               
              |
              <span>
                 
                <a
                  aria-label="Sort by Title"
                  class="filter-summary__sort-control"
                  href="/?s=Title"
                >
                  Title
                </a>
                 
              </span>
            </p>
          </div>
          <div
            class="tableWrapper"
          >
            <table
              aria-describedby="filter-summary"
              class="table"
              id="results"
            >
              <caption
                class="visually-hidden"
              >
                Guidance and quality standards in consultation
              </caption>
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    Title
                  </th>
                  <th
                    scope="col"
                  >
                    Consultation
                  </th>
                  <th
                    scope="col"
                  >
                    Type
                  </th>
                  <th
                    scope="col"
                  >
                    Consultation end date
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ng10160/consultation/html-content"
                    >
                      Type 2 diabetes in adults: management (update)
                    </a>
                  </td>
                  <td>
                    Draft guidance consultation
                  </td>
                  <td>
                    NICE guideline
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="14/10/2021"
                      datetime="2021-10-14T05:00:00"
                    >
                      <span>
                        14 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ipg10181/consultation/html-content"
                    >
                      Microwave ablation for treating primary lung cancer and metastases in the lung
                    </a>
                  </td>
                  <td>
                    Interventional procedure consultation
                  </td>
                  <td>
                    Interventional procedures guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="18/10/2021"
                      datetime="2021-10-18T05:00:00"
                    >
                      <span>
                        18 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ipg10179/consultation/html-content"
                    >
                      Stereotactic radiosurgery for trigeminal neuralgia
                    </a>
                  </td>
                  <td>
                    Interventional procedure consultation
                  </td>
                  <td>
                    Interventional procedures guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="18/10/2021"
                      datetime="2021-10-18T05:00:00"
                    >
                      <span>
                        18 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ipg10175/consultation/html-content"
                    >
                      Transcutaneous electrical stimulation of the supraorbital nerve for treating and preventing migraine
                    </a>
                  </td>
                  <td>
                    Interventional procedure consultation
                  </td>
                  <td>
                    Interventional procedures guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="18/10/2021"
                      datetime="2021-10-18T05:00:00"
                    >
                      <span>
                        18 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ta10478/consultation/html-content-3"
                    >
                      Venetoclax with a hypomethylating agent for untreated acute myeloid leukaemia when intensive chemotherapy is unsuitable [ID1564]
                    </a>
                  </td>
                  <td>
                    Appraisal consultation
                  </td>
                  <td>
                    Technology appraisal guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="25/10/2021"
                      datetime="2021-10-25T05:00:00"
                    >
                      <span>
                        25 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ta10556/consultation/html-content-2"
                    >
                      Ponesimod for treating relapsing multiple sclerosis [ID1393]
                    </a>
                  </td>
                  <td>
                    Appraisal consultation
                  </td>
                  <td>
                    Technology appraisal guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="26/10/2021"
                      datetime="2021-10-26T05:00:00"
                    >
                      <span>
                        26 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ng10140/consultation/html-content"
                    >
                      Mental wellbeing at work
                    </a>
                  </td>
                  <td>
                    Draft guidance consultation
                  </td>
                  <td>
                    NICE guideline
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="29/10/2021"
                      datetime="2021-10-29T05:00:00"
                    >
                      <span>
                        29 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ng10191/consultation/html-content"
                    >
                      Maternal and child nutrition
                    </a>
                  </td>
                  <td>
                    Draft scope consultation
                  </td>
                  <td>
                    NICE guideline
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="29/10/2021"
                      datetime="2021-10-29T05:00:00"
                    >
                      <span>
                        29 October 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ipg10163/consultation/html-content"
                    >
                      Intramedullary distraction for upper limb lengthening
                    </a>
                  </td>
                  <td>
                    Interventional procedure consultation
                  </td>
                  <td>
                    Interventional procedures guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="2/11/2021"
                      datetime="2021-11-02T05:00:00"
                    >
                      <span>
                        2 November 2021
                      </span>
                    </time>
                  </td>
                </tr>
                <tr>
                  <td>
                    <a
                      href="/guidance/gid-ipg10189/consultation/html-content"
                    >
                      Percutaneous insertion of a cystic duct stent after cholecystostomy for acute calculous cholecystitis
                    </a>
                  </td>
                  <td>
                    Interventional procedure consultation
                  </td>
                  <td>
                    Interventional procedures guidance
                  </td>
                  <td>
                    <time
                      class="date"
                      data-shortdate="2/11/2021"
                      datetime="2021-11-02T05:00:00"
                    >
                      <span>
                        2 November 2021
                      </span>
                    </time>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <nav
            aria-label="Pagination"
            class="pagination "
            role="navigation"
          >
            <ol
              class="pagination__list"
            >
              <li
                aria-current="true"
                class="pagination__item pagination__item--current"
              >
                <span
                  class="pagination__inactive"
                >
                  <span
                    class="visually-hidden"
                  >
                    Current page 
                  </span>
                  1
                </span>
              </li>
              <li
                class="pagination__item"
              >
                <a
                  aria-label="Go to page 2"
                  class="pagination__link"
                  href="/?pa=2"
                >
                  2
                </a>
              </li>
              <li
                class="pagination__item pagination__item--count"
              >
                <span>
                  Page 
                  <strong>
                    1
                  </strong>
                   of 
                  <strong>
                    2
                  </strong>
                </span>
              </li>
              <li
                class="pagination__item pagination__item--bookend"
              >
                <a
                  class="pagination__link"
                  href="/?pa=2"
                >
                  Next page
                </a>
              </li>
            </ol>
          </nav>
          <div
            class="grid grid--middle mt--d"
          >
            <div
              class="mb--e mb--0-sm"
              data-g="12 sm:6"
            >
              <button
                class="btn mv--0 mh--0"
                type="button"
              >
                Copy 
                10 results
                 to clipboard
              </button>
            </div>
            <div
              class="text-right"
              data-g="12 sm:6"
            >
              <h3
                class="p mt--0 mb--c"
              >
                Results per page
              </h3>
              <ol
                class="list list--piped mv--0"
              >
                <li
                  class="pl--b"
                >
                  10
                </li>
                <li
                  class="pl--b"
                >
                  <a
                    aria-label="Show 25 results per page"
                    class="ph--0"
                    href="/?ps=25"
                  >
                    25
                  </a>
                </li>
                <li
                  class="pl--b"
                >
                  <a
                    aria-label="Show 50 results per page"
                    class="ph--0"
                    href="/?ps=50"
                  >
                    50
                  </a>
                </li>
                <li
                  class="pl--b"
                >
                  <a
                    aria-label="Show All results per page"
                    class="ph--0"
                    href="/?ps=9999"
                  >
                    All
                  </a>
                </li>
              </ol>
            </div>
          </div>
        </section>
      </div>
    </div>
  </main>
  <noscript />
</div>
`;
